content/en/docs/tasks/observability/distributed-tracing/zipkin/test.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC1090,SC2154,SC2155,SC2034
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+source "tests/util/samples.sh"
+
+# @setup profile=demo
+
+## Setting up application
+# Set to known setting of sidecar injection
+kubectl label namespace default istio-injection=enabled --overwrite
+
+# ZIPKIN_MANIFEST_URL="https://raw.githubusercontent.com/istio/istio/release-1.7/samples/addons/extras/zipkin.yaml"
+kubectl apply -f "https://raw.githubusercontent.com/istio/istio/release-1.7/samples/addons/extras/zipkin.yaml"
+_wait_for_deployment istio-system zipkin
+
+# Install Bookinfo application
+startup_bookinfo_sample
+
+# This shows up in many places. Should move to util? 
+function send_productpage_requests() {
+  _set_ingress_environment_variables
+  local GATEWAY_URL="$INGRESS_HOST:$INGRESS_PORT"
+  for _ in {1..250}; do
+    curl -s "http://$GATEWAY_URL/productpage" > /dev/null
+  done
+}
```

Comment:
There is a snip for this, that isn't currently being generated by scripts/snip.py, because it doesn't currently handle boiler-plate.
```
{{< boilerplate trace-generation >}}
```
If handled properly, the snip generator would output a snip function, something like this:
```
snip_trace_generation() {
for i in `seq 1 100`; do curl -s -o /dev/null http://$GATEWAY_URL/productpage; done
}
``` 
If you would like to fix the snip generator to output boilerplate snips, that would be great. Otherwise, for now I would just simulate it by adding a fake function that looks like the expected snip and use it in the test
```suggestion
# TODO fix snip generator to gen this function in snips.sh
snip_trace_generation() {
for i in `seq 1 100`; do curl -s -o /dev/null http://$GATEWAY_URL/productpage; done
}
```

Reply:
Thanks. Let me check out that.

Reply:
Done. I generated snippets for boilerplate (of course, updated scripts/gen_snips.sh` and `scripts/snips.py`) and "sourced" it from `snip.sh` for each page. Reason - boilerplates are meant to be reusable. So it is better to have those snippets in one place and "source" them wherever needed. That's also why I used different prefix to avoid naming collision. Some minor changes are due to linter.